### PR TITLE
Make the i2c error render as text if eh1_0_alpha is on.

### PR DESCRIPTION
**Pull request by ithinuel**

### CODE REVIEW

File: rp2040-hal/src/i2c.rs [hunk 1]
```diff
@@ -80,6 +83,38 @@ pub enum Error {
     AddressReserved(u16),
 }
 
+#[cfg(feature = "eh1_0_alpha")]
+impl core::fmt::Debug for Error {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
```
Comment (9names) [inline rp2040-hal/src/i2c.rs:88]: Use the same name parameter name `fmt` for the formatter for both core::fmt::Formatter and defmt, no reason these should be different

**Review by 9names — CHANGES_REQUESTED** [review]

File: rp2040-hal/src/i2c.rs (hunk 1 above)
Comment (ithinuel) [reply rp2040-hal/src/i2c.rs:88]: The difference comes from how the parameter is called in the respective trait definitions. 
Any reason to chose `fmt` over `f` ?

Comment (9names) [reply rp2040-hal/src/i2c.rs:88]: No reason, just would prefer that it's the same one for consistency. 
I don't believe in making your project conform to the internal naming conventions of libraries.
Use f for both if you prefer brevity, fmt for both would be fine, as would formatter.

Comment (ithinuel) [reply rp2040-hal/src/i2c.rs]: opted for `fmt` as requested :)

**Review by 9names — APPROVED** [review]
Lgtm

**Review by jannic — APPROVED** [review]

**Pull request merged**